Makefile

```diff
@@ -63,7 +66,7 @@ sandbox-dev-up:
 
 sandbox-dev-stop:
```

Comment:
@ahangsu Unrelated to the line:  Gauging PR scope - Do you plan to apply a similar configuration change to readthedocs in-scope to this PR or a different PR?

* Specifically, I'm referring to https://docs.readthedocs.io/en/stable/config-file/v2.html#sphinx-fail-on-warning in `.readthedocs.yaml`.
* Either works for me provided we make the change prior to closing the story.

Reply:
I can make the change for `.readthedocs.yml` in this PR, but this made me somewhat worry about double config, tell me if I am paranoid: we have a `SPHINX_OPTION` in `Makefile`, and we (will) also have a `fail_on_warning: true` in `.readthedocs.yaml`.

If that is double config, I will remove the change in `Makefile` and only change in `.readthedocs.yaml`.

Reply:
Holdup, what I just said about double config might be wrong, here's my hunch:
- `bundle-docs` is just used for doc-test, right?
- then I felt necessary to also have `fail_on_warning: true` activated in `.readthedocs.yaml`, for that is on publishing, rather than testing purpose.

Reply:
@ahangsu Right - readthedocs build (https://readthedocs.org/projects/pyteal/builds/) happens out-of-band to our CI.  

So - We either want to:
* Make doc-test mirror the functionality of readthedocs build.
* Make readthedocs build part of our CI process.

Ideally, I _think_ we'd include readthedocs build as a status check to avoid duplicating build steps.  However, I haven't investigated it and I suspect it's more involved.  

For now, if we can make doc-test mirror readthedocs build, I think it's a worthwhile improvement.

Reply:
Thanks for explaining! I would more prefer including readthedocs build inside CI process. No wonder why readthedocs has only publish builds, and that confuses me yesterday.

For now, I am happy to let CI test mirror what is happening in documentation publishing, but we might need to open a future improvement ticket for readthedocs in CI test.

Reply:
if we would like to include readthedocs build inside CI, I think we will need to change github workflow somehow, and mimic https://github.com/algorand/py-algorand-sdk/blob/develop/.circleci/config.yml, where py-algorand-sdk uploads built docs to readthedocs under latest tag.

There might be some other factors I overlooked, let me know if that is the path we want to go.

Reply:
Just chiming in, the `docset` Circle CI step for the Python SDK does the same thing as PyTeal's [`build-docset`](https://github.com/algorand/pyteal/blob/d8b65b7231a45d33e951b5d867864c868e0109cb/.github/workflows/build.yml#L71) step. It's not responsible for publishing the docs to readthedocs.

Reply:
Thread status:  We agree _not_ integrating readthedocs build is the path of less resistance.

I'm leaving the thread open until an attempt is made to reproduce issue _fixed by_ https://github.com/algorand/pyteal/pull/472.

Reply:
@ahangsu Thanks for the experimentation - The build error in https://github.com/algorand/pyteal/pull/485 proves we'll catch a similar class of error in the future.  Resolving.